packages/running-extension/src/kernels.tsx

```diff
@@ -49,6 +58,59 @@ export async function addKernelRunningSessionManager(
   // Wait until the relevant services are ready.
   await Promise.all([kernels.ready, kernelspecs.ready, sessions.ready]);
 
+  async function getUnusedKernels() {
+    // Identifies unused kernels
+    const runningKernels = await KernelAPI.listRunning();
+    return runningKernels.filter(
+      kernel => kernel.connections !== undefined && kernel.connections < 1
+    );
+  }
+
+  async function checkShutdownUnusedEnabled() {
+    shutdownUnusedEnabled = (await getUnusedKernels()).length > 0;
+    commands.notifyCommandChanged(SHUTDOWN_UNUSED_BUTTON_CLASS);
```

Comment:
This needs to check whether the status changed before notifying, otherwise it would invoke throttler in an infinite loop (although throttled to once every 5 seconds) because `isEnabled` invokes it and it is invoked when the command changes. 

```suggestion
    const wasEnabled = shutdownUnusedEnabled;
    shutdownUnusedEnabled = (await getUnusedKernels()).length > 0;
    if (wasEnabled !== shutdownUnusedEnabled) {
      commands.notifyCommandChanged(SHUTDOWN_UNUSED_BUTTON_CLASS);
    }
```